doc/doxygen/headers/reordering.h

```diff
@@ -20,36 +20,90 @@
- * example, the first face of a triangle with nodes {0, 1, 2} is {0, 1}.
- * Similarly, the first face of a triangle with nodes {1, 0, 3} is {1, 0}. By
- * itself, this would be inconsistent, since the same line would be defined in
- * twice: once by each of the two adjacent cells, and they might not agree
- * whether the line should be {0, 1} or {1, 0}. To solve this problem deal.II
- * also stores an <tt>unsigned char</tt> on each cell for each of the cell's
- * faces; this value encodes the orientation of the line *as seen from this
- * cell*. In this particular case the orientation of the {1, 0} line from the
- * {0, 1, 2} cell is ReferenceCell::default_combined_face_orientation() whereas
- * the orientation of the {0, 1} line from the {1, 0, 3} cell is
- * ReferenceCell::reversed_combined_line_orientation().
+ * In 1D, lines are cells and, since each line appears in the Triangulation
+ * exactly once, they do not store any orientation information.
+ *
+ * A line in 2D is a face which may be shared by two cells, whereas in 3D a line
+ * may be shared by an arbitrary number of cells.
+ *
+ * In 2D, each cell stores the indices of the lines which bound it: i.e., each
+ * cell stores either 3 (for triangles) or 4 (for quadrilaterals) integers which
+ * enumerate those lines. Put another way: while there is no distinct `Line`
+ * class in deal.II, each line is uniquely identified by an index (accessed via
+ * CellAccessor::line_index()) and lines are, like other geometric entities,
+ * implemented with the flyweight pattern and represented by a TriaAccessor.
+ * Each line stores and is defined by the vertex indices which bound it and also
+ * stores other auxiliary information (such as boundary and manifold ids).
+ *
+ * In 3D, the line indices of a cell are stored by the faces which bound it:
+ * i.e., each cell stores its own face indices (which represent triangles or
+ * quadrilaterals) and each face stores its own line indices. Hence, when
+ * accessing a cell's lines, after identifying the face which owns that line
+ * (via ReferenceCell::standard_line_to_face_and_line_index()) all data lookups
+ * proceed in exactly the same way as the 2D case.
+ *
+ * Since each line has a unique index it also has a canonical vertex ordering.
+ * For example, consider two adjacent triangles with vertices `{0, 1, 2}` and
+ * `{1, 0, 3}`. The order of the vertices on each cell is defined by the
+ * CellData objects passed to Triangulation::create_triangulation(): typically,
+ * the order of the vertices is arbitrary aside from the constraint that they
+ * form a cell whose mapping to the reference cell has a positive Jacobian. In
+ * this example, the first face of the first cell is `{0, 1}` whereas the first
+ * face of the second cell is `{1, 0}`. In deal.II, the canonical order of a
+ * line's vertices is set by the first cell with that line: i.e., line `0` will
+ * have vertices `{0, 1}` since it first appears in the first cell.
```

Comment:
Is this how it is *defined*, or simply a consequence of the order in which these lines are created? In other words, is this something we can *promise*?